arelle/plugin/xbrlDB/XbrlSemanticJsonDB.py

```diff
@@ -130,6 +131,9 @@ def jsonDefaultEncoder(obj):
         return float(obj)
     elif isinstance(obj, (datetime.date, datetime.datetime)):
         return XmlUtil.dateunionValue(obj)
+    elif isinstance(obj, (IsoDuration, gYear, gMonthDay)):
+        return str(obj)
+
```

Comment:
Without this, parsed instance documents (a.k.a. source of facts) fail to encode to JSON (without raising an exception).